File: webui/src/lib/api/audit.ts
import { apiFetch } from './client';

export interface AuditEvent {
  id: number;
  action: string;
  description: string;
  ts: string;
  actor: string;
  sha256?: string | null;
  account_name?: string;
  filename?: string | null;
  reason?: string;
  details?: Record<string, unknown>;
}

export interface AuditPage {
  events: AuditEvent[];
  cursor: string | null;
  has_more: boolean;
}

export interface AuditDailySummary {
  day_utc: string;
  accepted_today: number;
  rejected_today: number;
}

export function getAuditLog(cursor?: string | null, limit = 50, action?: string | null): Promise<AuditPage> {
  const params = new URLSearchParams();
  params.set('limit', String(limit));
  if (cursor) {
    params.set('after', cursor);
  }
  if (action) {
    params.set('action', action);
  }
  return apiFetch<AuditPage>(`/api/v1/audit-log?${params.toString()}`);
}

export function getAuditDailySummary(): Promise<AuditDailySummary> {
  return apiFetch<AuditDailySummary>('/api/v1/audit-log/daily-summary');
}

File: api/schemas/audit.py
"""Pydantic schemas for audit endpoints."""

from __future__ import annotations

from pydantic import BaseModel


class AuditEvent(BaseModel):
    """Single audit log event."""

    id: int
    sha256: str | None = None
    account_name: str | None = None
    client_ip: str | None = None
    action: str
    description: str
    reason: str | None = None
    filename: str | None = None
    details: dict | None = None
    actor: str
    ts: str


class AuditPage(BaseModel):
    """Paginated list of audit events."""

    events: list[AuditEvent]
    cursor: str | None = None
    has_more: bool


class AuditDailySummary(BaseModel):
    """Current-day accepted/rejected outcome counts derived from audit events."""

    day_utc: str
    accepted_today: int
    rejected_today: int
